From 43e623f93ffb9386165c2149705ee716ef88e3a5 Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Fri, 8 Apr 2022 14:33:20 -0400
Subject: [PATCH 11/13] Fix: do not warn on unknown counter ioctl

It is perfectly valid for a newer lttng-tools to try to use an unknown
ioctl and handle -ENOSYS.

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Change-Id: Ia9f6472ca1196f983eee1327805b0ad69d028a98
---
 src/lttng-abi.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/lttng-abi.c b/src/lttng-abi.c
index fa7eca6f..c3e67690 100644
--- a/src/lttng-abi.c
+++ b/src/lttng-abi.c
@@ -730,7 +730,6 @@ long lttng_counter_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 		return lttng_kernel_counter_clear(counter, indexes);
 	}
 	default:
-		WARN_ON_ONCE(1);
 		return -ENOSYS;
 	}
 }
-- 
2.36.1

